flagutil/flags.go

```diff
@@ -0,0 +1,97 @@
+package flagutil
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+)
+
+func ValidateFlags(flagset *flag.FlagSet, args []string) error {
+	validator := flagValidator{
+		args:    args,
+		flagset: flagset,
+	}
+	for {
+		seen, err := validator.validateNextFlag()
+		if seen {
+			continue
+		}
+		if err == nil {
+			break
+		}
+		return err
+	}
+	return nil
+}
+
+type flagValidator struct {
+	args    []string
+	flagset *flag.FlagSet
+}
+
+// optional interface to indicate boolean flags that can be
+// supplied without "=value" text
+type boolFlag interface {
+	IsBoolFlag() bool
+}
+
+func (v *flagValidator) validateNextFlag() (bool, error) {
```

Comment:
This method is based on the `parseOne()` method from https://golang.org/src/flag/flag.go, but was rewritten for extra validation rather than parsing flag values.